my_game.py

```diff
@@ -374,7 +397,20 @@ def on_update(self, delta_time):
         # Player rocket engine
         if self.up_pressed:
             self.player_sprite.player_thrust()
-        
+
+            # Only make rocket_controller if one doesn't already exist
+            if self.player_rocket is None:
```

Comment:
Maybe we could always have the _self.player_rocket_, but have it with a controller which emits 0 particles? Then we can set a new controller when the thrust key is ressed?

Reply:
player_rocket has been renamed "player_rocket_emitter"

The emitter is not the one controlling particle stuff, so I tried propping the emitter code into the variable, but the game froze, so I assumed it wasn't quite the right thing to do, Elsa got angry every time (ha ha very fun pun, the game _froze_)